### ISSUE TITLE
Level up while Dead

### ISSUE THREAD

**Issue by pangolp** [azerothcore/azerothcore-wotlk#16516]
### What client do you play on?
deDE

### Faction
* [x]  Alliance

### Content Phase:
* [x]  Generic
* [x]  1-19

### Current Behaviour
I was play a warsong battleground and died. When i died i got a level up and got full hp and mana. Stand there where i died but could not do anything. After /Reload i was able to walk around but still not cast anything. Always got the error "you are dead"

* was not able to get in target (but they could see me)
* was not able to use spells or food ect.
* didnt got any exp
* could not get rezeed from ghosthealer (didnt see him)

after battleground all was fine again

### Expected Blizzlike Behaviour
i think you should not be able to get a level up while you are dead, as you dont get any exp when you are dead.

### Steps to reproduce the problem
1. create a character
2. kill him .die
3. give him a lvl up while he lay on ground

### Extra Notes
Already tested from Roboto from ChromieCraft - he asked me to open this bug.

### AC rev. hash/commit
[chromiecraft/azerothcore-wotlk@48fccc8](https://github.com/chromiecraft/azerothcore-wotlk/commit/48fccc851d8491c9019a585c74417641376795de)

### Operating system
Ubuntu 20.04

### Modules
* [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
* [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
* [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
* [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
* [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
* [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
* [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
* [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
* [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
* [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
* [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
* [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
* [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
* [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
* [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
* [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
* [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
* [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
* [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
* [mod-transmog](https://github.com/azerothcore/mod-transmog)
* [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
* [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
* [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
* [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
* [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
* [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
* [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
* [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
* [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations
None

### Server
ChromieCraft

**Comment by pangolp** [comment]
@AzerothCoreDev If they can remove the hash tag, so much the better. Because I created a report just like that, so I can edit it later.

**Comment by pangolp** [comment]
I have tried to replicate this case, but the only way in which it is fulfilled, and if through a GM command, is to use the `.character level` or the `.levelup`. Otherwise, I have not been able to replicate it. It could be added, in both commands, that if the character is dead, the command cannot be used on him, but I don't know if it is really necessary to do that check. It would be more in the hands of the game masters, not using the commands, in dead players.

**Comment by pangolp** [comment]
I want to clarify, just in case, that I did not test it on a battlefield, I tested it in the open world, doing missions, trying to see if, when being in a group, and one of the characters died, he would continue to obtain a level, to I can force it to pass the level and thus revive it, reproducing the error described here, but I couldn't do that. On the battlefield, or dungeon, I didn't try it.

**Comment by laasker** [comment]
> I want to clarify, just in case, that I did not test it on a battlefield, I tested it in the open world, doing missions, trying to see if, when being in a group, and one of the characters died, he would continue to obtain a level, to I can force it to pass the level and thus revive it, reproducing the error described here, but I couldn't do that. On the battlefield, or dungeon, I didn't try it.

I was being able to replicate it in BG. While one character was dead, i delivered the flag and the dead character leveled up and got resurrected

**Comment by amed80** [comment]
I don't know if you are suppose to level up and get resurrected but as far of gaining experience while dead in battleground , watching [this video](https://youtu.be/X201oz25Uh0?t=2015) from wotlk classic , at 33;37 you can see that he gained experience while dead/ghost.
Also this [reddit](https://www.reddit.com/r/classicwow/comments/y5k67b/insane_wotlk_bg_leveling_1milhr/) post , watch the video at 0:39 the guy says you still gain experience while dead in AV.

**Comment by pangolp** [comment]
> I don't know if you are suppose to level up and get resurrected but as far of gaining experience while dead in battleground , watching [this video](https://youtu.be/X201oz25Uh0?t=2015) from wotlk classic , at 33;37 you can see that he gained experience while dead/ghost.
> Also this [reddit](https://www.reddit.com/r/classicwow/comments/y5k67b/insane_wotlk_bg_leveling_1milhr/) post , watch the video at 0:39 the guy says you still gain experience while dead in AV.

I just read this comment. I really thought it couldn't be reproduced, but it seems there are some ways for this report to be valid, so it should remain open.

**Comment by alecsci** [comment]
@pangolp please update the issue title, you can use mine, that's referenced by @laasker

**Comment by heyitsbench** [comment]
No need to update the title if the issue is more generic than just dying in a battleground.

**Comment by alecsci** [comment]
@heyitsbench he can provide a better and generic description, instead of the existing one.
Seems like this issue, should address 2 problems: level up while dead, and the character being stuck if dead at prep time.

**Comment by heyitsbench** [comment]
My contention is that this issue isn't limited to battlegrounds, players on CC have reported this issue happening while doing dungeons as well. The linked issue is *very* specific to battlegrounds, this one is a bit more generic. Though they both seem to share symptoms, I do not believe fixing one would inherently fix both.

**Comment by pangolp** [comment]
When I did the tests at the time, it was in the open world. The character died, gained experience, leveled up and in doing so, I regained my whole life, but I couldn't move. By disconnecting and reconnecting, the character in the selection menu was dead and when I entered, I was indeed dead. That's all I remember about that problem and that's why I believe the report at the time, based on a report from ChromieCraft if I remember correctly. The weird thing is that in the description, I don't link it to any issue. But I'm sure I got it from ChromieCraft's bugtracker.
